apps/src/netsim/NetSimRouterNode.js

```diff
@@ -1051,11 +1116,53 @@ NetSimRouterNode.prototype.updateRouterQueue_ = function (rows) {
   }
 
   this.routerQueueCache_ = newQueue;
-  // TODO: Drop packets that exceed queue memory (IF not already processing?)
   this.recalculateSchedule();
+  this.enforceMemoryLimit_();
+
   // Propagate notification of queue change (for stats, etc)
 };
 
+NetSimRouterNode.prototype.enforceMemoryLimit_ = function () {
+  if (this.enforcingMemoryLimit_) {
+    return;
+  }
+  // Just drop the first packet that we can simulate that's beyond the memory
+  // limit
+  var droppablePacket = this.findFirstLocallySimulatedPacketOverMemoryLimit();
+  if (droppablePacket) {
+    this.enforcingMemoryLimit_ = true;
+    this.removeRowFromSchedule_(droppablePacket);
+    var droppableMessage = new NetSimMessage(this.shard_, droppablePacket);
+    droppableMessage.destroy(function (err) {
+      if (err) {
+        this.enforcingMemoryLimit_ = false;
+        return;
+      }
+
+      this.log(droppableMessage.payload, NetSimLogEntry.LogStatus.DROPPED);
+      this.enforcingMemoryLimit_ = false;
+    }.bind(this));
+  }
+};
+
+/**
+ * Walk the router queue, and return the first packet we find beyond the router's
+ * memory capacity that the local simulation controls and is able to drop.
+ * @returns {messageRow|undefined} undefined if no such message is found.
+ */
+NetSimRouterNode.prototype.findFirstLocallySimulatedPacketOverMemoryLimit = function () {
+  var packet;
+  var usedMemory = 0;
+  for (var i = 0; i < this.routerQueueCache_.length; i++) {
+    packet = this.routerQueueCache_[i];
+    usedMemory += packet.payload.length;
+    if (usedMemory > this.memory && this.localSimulationOwnsMessageRow_(packet)) {
+      return packet;
+    }
+  }
+  return undefined;
```

Comment:
Interesting. I think I prefer null, but don't feel all that strongly about it so I'd say stick with whichever feels more right to you :)